snuba/migrations/snuba_migrations/metrics/0007_granularity_10s.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import AggregateFunction, Column, Float, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.snuba_migrations.metrics.templates import (
+    get_forward_view_migration_local,
+)
+
+
+def drop_views() -> List[operations.SqlOperation]:
+    return [
+        operations.DropTable(
+            storage_set=StorageSetKey.METRICS,
+            table_name=f"metrics_{metric_type}_mv_local",
+        )
+        for metric_type in ("sets", "counters", "distributions")
+    ]
+
+
+def recreate_views(granularity: int) -> List[operations.SqlOperation]:
+    return [
+        get_forward_view_migration_local(
+            source_table_name="metrics_buckets_local",
+            table_name="metrics_sets_local",
+            mv_name="metrics_sets_mv_local",
+            aggregation_col_schema=[
+                Column("value", AggregateFunction("uniqCombined64", [UInt(64)])),
+            ],
+            aggregation_states="uniqCombined64State(arrayJoin(set_values)) as value",
+            granularity=granularity,
+        ),
+        get_forward_view_migration_local(
+            source_table_name="metrics_counters_buckets_local",
+            table_name="metrics_counters_local",
+            mv_name="metrics_counters_mv_local",
+            aggregation_col_schema=[
+                Column("value", AggregateFunction("sum", [Float(64)])),
+            ],
+            aggregation_states="sumState(value) as value",
+            granularity=granularity,
+        ),
+        get_forward_view_migration_local(
+            source_table_name="metrics_distributions_buckets_local",
+            table_name="metrics_distributions_local",
+            mv_name="metrics_distributions_mv_local",
+            aggregation_col_schema=[
+                Column(
+                    "percentiles",
+                    AggregateFunction(
+                        "quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]
+                    ),
+                ),
+                Column("min", AggregateFunction("min", [Float(64)])),
+                Column("max", AggregateFunction("max", [Float(64)])),
+                Column("avg", AggregateFunction("avg", [Float(64)])),
+                Column("sum", AggregateFunction("sum", [Float(64)])),
+                Column("count", AggregateFunction("count", [Float(64)])),
+            ],
+            aggregation_states=(
+                "quantilesState(0.5, 0.75, 0.9, 0.95, 0.99)((arrayJoin(values) AS values_rows)) as percentiles, "
+                "minState(values_rows) as min, "
+                "maxState(values_rows) as max, "
+                "avgState(values_rows) as avg, "
+                "sumState(values_rows) as sum, "
+                "countState(values_rows) as count"
+            ),
```

Comment:
There is quite some code duplication here. We could define the schema and states as constants somewhere, but I did not want to create a larger diff than necessary in the existing migrations.